### PR TITLE
Bug fix 3.10/proper namespacing for gtests

### PR DESCRIPTION
**Pull request by dothebart**
backport of https://github.com/arangodb/arangodb/pull/18725

### CODE REVIEW

**Review by mpoeter — APPROVED** [review]

**Review by KVS85 — APPROVED** [review]

File: js/client/modules/@arangodb/testsuites/gtest.js [hunk 1]
```diff
@@ -112,19 +112,33 @@ function getGTestResults(fileName, defaultResults) {
   return results;
 }
 
-function gtestRunner (testname, options) {
+function gtestRunner (testfilename, name, opts, testoptions) {
+  let options = _.clone(opts);
```
Comment (alexbakharew) [inline js/client/modules/@arangodb/testsuites/gtest.js:116]: Bettre to use `cloneDeep()` ?

File: js/client/modules/@arangodb/testsuites/gtest.js [hunk 2]
```diff
@@ -136,37 +150,43 @@ function gtestRunner (testname, options) {
...
+    testFns[test] = x => gtestRunner(test, test, x);
+  }
+  testFns['gtest'] = x => gtestRunner('arangodbtests', 'arangodbtests', x);
+
+  testFns['gtest_replication2'] = x => gtestRunner('arangodbtests_replication2', 'gtest-replication2', x);
+  
+  let iresearch_filter = ['gtest_filter=IResearch*'];
+  testFns['gtest_iresearch'] = x => gtestRunner('arangodbtests', 'gtest-iresearch', x, iresearch_filter);
+  let no_iresearch_filter = ['gtest_filter=-IResearch*:LongRunning*'];
+  testFns['gtest_arangodb'] = x => gtestRunner('arangodbtests', 'gtest-arangodb', x, no_iresearch_filter);
```
Comment (alexbakharew) [inline js/client/modules/@arangodb/testsuites/gtest.js:189]: I think that all test cases can be located in `const tests` array?

File: js/client/modules/@arangodb/testsuites/gtest.js (hunk 1 above)
Comment (dothebart) [reply js/client/modules/@arangodb/testsuites/gtest.js:116]: options is one-dimensional (see testing.js)

File: js/client/modules/@arangodb/testsuites/gtest.js (hunk 2 above)
Comment (dothebart) [reply js/client/modules/@arangodb/testsuites/gtest.js:189]: tests iterates over binaries to run; we want to have different options. but yes, could probably be done - but rather later for devel and not for the backport.

**Pull request merged**